### PR TITLE
Add support for changing over bookmark paths using the legacy bookmak path

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Though the converter would carry over the bookmark this was still not in the data path directory. This converts the path from static to flow properly.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: internal/converter/internal/staticconvert/testdata/sanitize.river [hunk 1]
```diff
@@ -42,13 +42,13 @@ loki.relabel "logs_integrations_integrations_windows_exporter_application" {
...
 	eventlog_name          = "Application"
 	xpath_query            = ""
-	bookmark_path          = "C:\\grafana_test\\Grafana Agent\\bookmarks.xml"
 	poll_interval          = "0s"
 	use_incoming_timestamp = true
 	forward_to             = [loki.relabel.logs_integrations_integrations_windows_exporter_application.receiver]
 	labels                 = {
 		job = "integrations/windows_exporter",
 	}
+	legacy_bookmark_path = "C:\\grafana_test\\Grafana Agent\\bookmarks.xml"
```
Comment (mattdurham) [inline internal/converter/internal/staticconvert/testdata/sanitize.river:51]: @erikbaranowski for some reason this did not like having anymore than one space after the name and the equals sign.

File: docs/sources/flow/reference/components/loki.source.windowsevent.md [hunk 2]
```diff
@@ -32,26 +32,29 @@ log entries to the list of receivers passed in `forward_to`.
...
+`exclude_event_data`     | `bool`               | Exclude event data.                                                                 | `false`                    | no
+`exclude_user_data`      | `bool`               | Exclude user data.                                                                  | `false`                    | no
+`exclude_event_message`  | `bool`               | Exclude the human-friendly event message.                                           | `false`                    | no
+`use_incoming_timestamp` | `bool`               | When false, assigns the current timestamp to the log when it was processed.         | `false`                    | no
+`forward_to`             | `list(LogsReceiver)` | List of receivers to send log entries to.                                           |                            | yes
+`labels`                 | `map(string)`        | The labels to associate with incoming logs.                                         |                            | no 
+`legacy_bookmark_path`   | `string`             | The location of the static bookmark path, will convert to flow compatible bookmark. | ``                         | no
```
Comment (erikbaranowski) [inline docs/sources/flow/reference/components/loki.source.windowsevent.md:48]: I think we want to say something here beyond `static bookmark path` which may not be obvious on what that means. Maybe a short description is needed up here and a more complete description down below? WDYT?

**Review by erikbaranowski — APPROVED** [review]
Some wording nits and code nits to address what you feel is necessary, but looks good!

File: internal/converter/internal/staticconvert/testdata/sanitize.river (hunk 1 above)
Comment (erikbaranowski) [reply internal/converter/internal/staticconvert/testdata/sanitize.river:51]: specifically here for static mode is where it gets called: https://github.com/grafana/agent/blob/b827bad63156fad4033f1c2ab67d285086bd157f/internal/converter/internal/staticconvert/staticconvert.go#L59

File: docs/sources/flow/reference/components/loki.source.windowsevent.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/loki.source.windowsevent.md:48]: We also should refer to Grafana Agent Flow in the same way... ie `flow` should be `{{< param "PRODUCT_NAME" >}}`

File: docs/sources/flow/reference/components/loki.source.windowsevent.md [hunk 3]
```diff
@@ -32,26 +32,29 @@ log entries to the list of receivers passed in `forward_to`.
...
 
 
 > **NOTE**: `eventlog_name` is required if `xpath_query` does not specify the event log.
 > You can define `xpath_query` in [short or xml form](https://docs.microsoft.com/en-us/windows/win32/wes/consuming-events).
 > When using the XML form you can specify `event_log` in the `xpath_query`.
 > If using short form, you must define `eventlog_name`.
 
+> **NOTE**: `legacy_bookmark_path` will only convert the bookmark if `bookmark_path` does not exist.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.source.windowsevent.md:56]:
```suggestion
{{< admonition type="note" >}}
`legacy_bookmark_path` will only convert the bookmark if `bookmark_path` does not exist.
{{< /admonition >}}
```

File: internal/component/loki/source/windowsevent/component_windows.go [hunk 4]
```diff
@@ -110,18 +110,30 @@ func (c *Component) Update(args component.Arguments) error {
 		newArgs.BookmarkPath = path.Join(c.opts.DataPath, "bookmark.xml")
 	}
 
-	// Create the bookmark file and parent folders if they don't exist.
+	// Check to see if we need to move the legacy bookmark path.
```
[the remark this answers is not in the document]
Comment (mattdurham) [reply internal/component/loki/source/windowsevent/component_windows.go:113]: That was a dangling comment, reworded it and moved the code into its own function.

**Pull request merged**